kafka/writer.go

```diff
@@ -55,12 +117,78 @@ func (w *Writer) Write(key string, req *remote.WriteRequest) error {
 	if err != nil {
 		return err
 	}
-	// set key to "${job}::${instance}" so that messages from the same job-instance consistently
-	// go to the same kafka partition
-	_, _, err = w.producer.SendMessage(&sarama.ProducerMessage{
+
+	m := &sarama.ProducerMessage{
 		Topic: w.topic,
 		Key:   sarama.StringEncoder(key),
 		Value: sarama.ByteEncoder(data),
+	}
+
+	go func() {
+		w.producer.Input() <- m
+
+		w.enqueued++
```

Comment:
whops totally forgot.  but i guess irrelevant since we're not those counters will be removed now.